lib/graphql/input_object_type.rb

```diff
@@ -51,6 +51,18 @@ def kind
     def validate_non_null_input(input, warden)
       result = GraphQL::Query::InputValidationResult.new
 
+      if input.is_a? Array
```

Comment:
Agreed `is_a?(Hash)` is not a great solution. Rails 5 params fail that check, right? 

How does this approach compare to something like `rescue NoMethodError` ? Is there a viable solution that doesn't make assumptions about the input's inheritance chain? :thinking: